### PR TITLE
Provide GIT_DESCRIBE token

### PR DESCRIPTION
**Pull request by szymex**
Provide result of 'git describe' as a token macro: GIT_DESCRIBE.

### CODE REVIEW

File: src/main/java/hudson/plugins/git/GitDescribeTokenMacro.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2011 CloudBees, Inc.
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitDescribeTokenMacro.java]: ???

File: src/main/resources/hudson/plugins/git/GitDescribeTokenMacro/help.jelly [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+  <dt>$${GIT_DESCRIBE}</dt>
+  <dd>
+    Expands to the description of a commit using the most recent tag reachable from it.
+  </dd>
+</j:jelly>
```
Comment (KostyaSha) [inline src/main/resources/hudson/plugins/git/GitDescribeTokenMacro/help.jelly]: use static html

**Comment by KostyaSha** [comment]
git describe has many options, do you plan support them?

**Comment by szymex** [comment]
In a future, we could add parameters like '--abbrev=<n>', '--all',  '--dirty'

File: src/main/java/hudson/plugins/git/GitSCM.java [hunk 3]
```diff
@@ -1063,6 +1064,13 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
             throw new IOException("Could not checkout " + revToBuild.revision.getSha1String(), e);
         }
 
+        //read ´git describe´
+        try {
+            environment.put(GIT_DESCRIBE, git.describe(revToBuild.revision.getSha1String()));
+        } catch (GitException e) {
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitSCM.java]: Why do you need put it as env variable if you already implemented it in a token?

**Comment by KostyaSha** [comment]
AFAIK token resolves variable directly when it called, i see no sense in having token that resolves through variable.

**Comment by szymex** [comment]
OK, could we then agree only on having variable?

**Comment by MarkEWaite** [comment]
I don't understand your question:

> OK, could we then agree only on having variable?

I thought that @KostyaSha was saying that if you have a token macro, it is then available in the same contexts as an environment variable, but with the added flexibility of parameter evaluation.  Can you test drive the change without making it an environment variable?

**Comment by szymex** [comment]
Problem is I do not really know, how to read git describe from GitDescribeTokenMacro class. How to instantiate GitClient instance.

**Comment by szymex** [comment]
I will refactor solution, without environment variable.

**Pull request closed without merging**